src/pymorize/std_lib/__init__.py

```diff
@@ -184,14 +189,24 @@ def temporal_resample(
     - 'D': daily
     - 'H': hourly
 
-    The resampling method (mean, sum, etc.) is determined by the rule.
+    See Also
+    --------
+    https://docs.xarray.dev/en/stable/user-guide/time-series.html#resampling-and-grouped-operations
     """
     if not has_time_axis(data):
         return data
 
     time_dim = get_time_label(data)
     freq = rule.data_request_variable.frequency
-    return data.resample({time_dim: freq}).mean()
+    if not freq_is_coarser_than_data(freq, data):
+        raise PymorizeResamplingTimeAxisIncompatibilityError(
+            f"Requested frequency {freq} is finer than the dataset's inherent frequency. Cannot resample!"
+        )
+    try:
+        return data.resample({time_dim: freq}).mean()
+    except Exception as e:
+        logger.exception(e)
+        raise PymorizeResamplingError(f"Error during resampling: {e}")
```

Comment:
Same as above, which variable in the data is affected by this?

Reply:
```suggestion
        raise PymorizeResamplingError(f"Error during resampling model {rule.model_variable} for CMOR {rule.cmor_variable}: {e}")
```